addons/account_edi_proxy_client/models/account_edi_format.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+from odoo import models
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    # -------------------------------------------------------------------------
+    # Helpers
+    # -------------------------------------------------------------------------
+
+    def _get_proxy_user(self, company):
+        '''Returns the proxy_user associated with this edi format.
+        '''
+        self.ensure_one()
+        return company.account_edi_proxy_client_ids.filtered(lambda u: u.edi_format_code == self.code)
+
+    def _create_proxy_user(self, company):
+        self.ensure_one()
+        edi_identification = self._get_proxy_identification(company)
+        if not edi_identification:
+            return
+
+        self.env['account_edi_proxy_client.user'].register_proxy_user(company, self, edi_identification)
+
+    # -------------------------------------------------------------------------
+    # To override
+    # -------------------------------------------------------------------------
+
+    def _get_proxy_identification(self, company):
```

Comment:
seems overkill.  Not everything should be made abstract.

Reply:
But it is. Each edi_format must have some kind of proxy_identification because this is what the identify the user uniquely relating with the edi_format.

Reply:
@bfr-o Could not we just use the edi_identification field anyways?  (maybe with PEPPOL we would need an additional field to say which identification type we use?)